### DISCUSSION TITLE
支持文件输入输出

### DISCUSSION THREAD

**Discussion by lnw143** [cpeditor/cpeditor-cn#32] [提问] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is this feature provided in the master branch?

- [X] I have checked the changelog

### The problem you are facing

当我使用`freopen`输入输出到文件时，CP Editor似乎并不支持空的输入

```
 [运行器] [所有输入均为空，没有需要运行的]
```

当我在输入框中输入一些内容后，它并不会真的运行程序，输出文件中仍没有内容

测试代码：
```cpp
#include<cstdio>
using namespace std;
int main() {
	freopen(".in","r",stdin);
	freopen(".out","w",stdout);
	int a,b;
	scanf("%d%d",&a,&b);
	printf("%d",a+b);
	return 0;
}
```

### Describe the feature you'd like

- 支持文件输入输出
- 测试输入输出可以打包指定文件为指定的输入文件名，如`a.in`为程序的读入文件，测试用例在`a01.in`中，我希望CP Editor能够将`a01.in`中的内容拷贝到程序目录下的`a.in`中并比较`.out`文件与`a01.ans`，实现自动化测试

### Are you willing to contribute to this?

- [ ] I'm ready to contribute to this (and will open a PR soon)
- [ ] I'd like to have a try (with the help of the maintainers)
- [X] No, thanks

**Comment by ouuan** [comment]
你应当不需要这个功能，“从文件加载成对的测试点”“从文件中加载”“保存到文件”等功能可以满足一般的需求。

可以使用 `#ifdefine` 来在 CP Editor 中不使用 `freopen`。

**Reply by lx0311** [reply]
样例太大，添加不了，必须用freopen怎么办

**Reply by ouuan** [reply]
> 样例太大，添加不了，必须用freopen怎么办

请问“添加不了”的具体表现是什么

**Reply by lx0311** [reply]
添加之后就卡爆了

**Reply by ouuan** [reply]
你的测例有多大，CP Editor 设置里长度限制都设的多少，电脑内存有多大？

**Reply by lx0311** [reply]
测例6836KB
长度999999999

**Reply by lx0311** [reply]
不能用freopen吗

**Reply by ouuan** [reply]
> 测例6836KB
> 长度999999999

哪个长度？但不管哪个都太大了，这个限制就是为了不卡死

**Reply by lx0311** [reply]
这样也很卡
![{59C802A1-7B33-4265-81B5-F2A3AFA02431}](https://github.com/cpeditor/cpeditor-cn/assets/118534434/5f3e1978-9ba0-44a2-bb68-70baf6f66ad6)

**Reply by ouuan** [reply]
你看看默认值是多少..

**Reply by lx0311** [reply]
调成默认值后又这样了
![Uploading QQ截图20231108114132.png…]()

**Comment by ouuan** [comment]
只不过有个问题是把测例存下来可能会重复占用空间，保存 session 的时候也可能很大 🤔

**Comment by ouuan** [comment, marked as answer]
现在也是可以 freopen 的，但是相对路径是相对于可执行文件的，默认是临时文件，可以在设置里修改，也可以右键标签页查看位置；也可以用绝对路径。

**Reply by lx0311** [reply]
谢谢

**Comment by wdragen** [comment]
从文件加载testcase的时候，可以这样改进，没有必要把文件内容加载到界面，界面只用显示文件的路径就ok，点击路径可以弹窗查看文件
@ouuan
